src/angular/components/set-password.component.ts

```diff
@@ -29,11 +30,17 @@ export class SetPasswordComponent extends BaseChangePasswordComponent {
     constructor(i18nService: I18nService, cryptoService: CryptoService, messagingService: MessagingService,
         userService: UserService, passwordGenerationService: PasswordGenerationService,
         platformUtilsService: PlatformUtilsService, policyService: PolicyService, private router: Router,
-        private apiService: ApiService) {
+        private apiService: ApiService, private syncService: SyncService) {
         super(i18nService, cryptoService, messagingService, userService, passwordGenerationService,
             platformUtilsService, policyService);
     }
 
+    async ngOnInit() {
+        this.formPromise = this.syncService.fullSync(true);
+        await this.formPromise;
```

Comment:
This isn't related to the UI form so I would not be assigning it to the form promise. We might need a `loading` state with a spinner for this now, though, since the UI will not be in a valid state until the sync completes.

Reply:
Updated with tracking boolean.